### PR TITLE
K8SPS-377: do not use --skip-ssl for db init

### PR DESCRIPTION
**Pull request by pooknull**
[K8SPS-377](https://jira.percona.com/browse/K8SPS-377) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://perconadev.atlassian.net/browse/K8SPS-377

**DESCRIPTION**
---
This PR removes the `--skip-ssl` option for database initialization

**CHECKLIST**
---
**Jira**
- [x] Is the Jira ticket created and referenced properly?
- [x] Does the Jira ticket have the proper statuses for documentation (`Needs Doc`) and QA (`Needs QA`)?
- [x] Does the Jira ticket link to the proper milestone (Fix Version field)?

**Tests**
- [ ] Is an E2E test/test case added for the new feature/change?
- [ ] Are unit tests added where appropriate?

**Config/Logging/Testability**
- [x] Are all needed new/changed options added to default YAML files?
- [x] Did we add proper logging messages for operator actions?
- [x] Did we ensure compatibility with the previous version or cluster upgrade process?
- [x] Does the change support oldest and newest supported PS version?
- [x] Does the change support oldest and newest supported Kubernetes version?

[K8SPS-377]: https://perconadev.atlassian.net/browse/K8SPS-377?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/version-service.log) |
| async-ignore-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/async-ignore-annotations.log) |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/auto-config.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/config.log) |
| config-router | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/config-router.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/demand-backup.log) |
| gr-demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/gr-demand-backup.log) |
| gr-demand-backup-haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/gr-demand-backup-haproxy.log) |
| gr-finalizer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/gr-finalizer.log) |
| gr-haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/gr-haproxy.log) |
| gr-ignore-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/gr-ignore-annotations.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/gr-init-deploy.log) |
| gr-one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/gr-one-pod.log) |
| gr-recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/gr-recreate.log) |
| gr-scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/gr-scaling.log) |
| gr-security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/gr-security-context.log) |
| gr-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/gr-self-healing.log) |
| gr-tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/gr-tls-cert-manager.log) |
| gr-users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/gr-users.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/haproxy.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/init-deploy.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/limits.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/monitoring.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/one-pod.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/operator-self-healing.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/recreate.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/scaling.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/service-per-pod.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/sidecars.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/smart-update.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/tls-cert-manager.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/24a399ea/users.log) |
| We run 32 out of 32|

commit: https://github.com/percona/percona-server-mysql-operator/pull/723/commits/24a399ea986b8c41f4017a4d60f0c58da581472d
image: `perconalab/percona-server-mysql-operator:PR-723-24a399ea`

File: build/ps-entrypoint.sh [hunk 1]
```diff
@@ -226,7 +227,14 @@ if [ "$1" = 'mysqld' -a -z "$wantHelp" ]; then
 		echo 'Initializing database'
 		# we initialize database into $TMPDIR because "--initialize-insecure" option does not work if directory is not empty
 		# in some cases storage driver creates unremovable artifacts (see K8SPXC-286), so $DATADIR cleanup is not possible
-		"$@" --initialize-insecure --skip-ssl --datadir="$TMPDIR"
+		init_args=(--initialize-insecure --datadir="$TMPDIR")
+		# --skip-ssl was removed in 8.4
+		if [ "$MYSQL_MINOR_VERSION" -lt 4 ]; then
```
Comment (hors) [inline build/ps-entrypoint.sh:232]: @egegunes @pooknull, maybe we can remove it for all versions ...

Comment (egegunes) [reply build/ps-entrypoint.sh:232]: I agree

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| version-service | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/version-service.log) |
| async-ignore-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/async-ignore-annotations.log) |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/auto-config.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/config.log) |
| config-router | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/config-router.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/demand-backup.log) |
| gr-demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-demand-backup.log) |
| gr-demand-backup-haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-demand-backup-haproxy.log) |
| gr-finalizer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-finalizer.log) |
| gr-haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-haproxy.log) |
| gr-ignore-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-ignore-annotations.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-init-deploy.log) |
| gr-one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-one-pod.log) |
| gr-recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-recreate.log) |
| gr-scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-scaling.log) |
| gr-scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-scheduled-backup.log) |
| gr-security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-security-context.log) |
| gr-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-self-healing.log) |
| gr-tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-tls-cert-manager.log) |
| gr-users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-users.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/haproxy.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/init-deploy.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/limits.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/monitoring.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/one-pod.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/operator-self-healing.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/recreate.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/scheduled-backup.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/service-per-pod.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/sidecars.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/smart-update.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/tls-cert-manager.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/users.log) |
| We run 34 out of 34|

commit: https://github.com/percona/percona-server-mysql-operator/pull/723/commits/613f0f8e3372fe65f4c66a57eceec4491c7017d2
image: `perconalab/percona-server-mysql-operator:PR-723-613f0f8e`

**Review by hors — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/version-service.log) |
| async-ignore-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/async-ignore-annotations.log) |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/auto-config.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/config.log) |
| config-router | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/config-router.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/demand-backup.log) |
| gr-demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-demand-backup.log) |
| gr-demand-backup-haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-demand-backup-haproxy.log) |
| gr-finalizer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-finalizer.log) |
| gr-haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-haproxy.log) |
| gr-ignore-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-ignore-annotations.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-init-deploy.log) |
| gr-one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-one-pod.log) |
| gr-recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-recreate.log) |
| gr-scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-scaling.log) |
| gr-scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-scheduled-backup.log) |
| gr-security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-security-context.log) |
| gr-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-self-healing.log) |
| gr-tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-tls-cert-manager.log) |
| gr-users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/gr-users.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/haproxy.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/init-deploy.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/limits.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/monitoring.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/one-pod.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/operator-self-healing.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/recreate.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/scheduled-backup.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/service-per-pod.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/sidecars.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/smart-update.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/tls-cert-manager.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-723/613f0f8e/users.log) |
| We run 34 out of 34|

commit: https://github.com/percona/percona-server-mysql-operator/pull/723/commits/613f0f8e3372fe65f4c66a57eceec4491c7017d2
image: `perconalab/percona-server-mysql-operator:PR-723-613f0f8e`

File: build/ps-entrypoint.sh [hunk 2]
```diff
@@ -226,7 +226,7 @@ if [ "$1" = 'mysqld' -a -z "$wantHelp" ]; then
 		echo 'Initializing database'
 		# we initialize database into $TMPDIR because "--initialize-insecure" option does not work if directory is not empty
 		# in some cases storage driver creates unremovable artifacts (see K8SPXC-286), so $DATADIR cleanup is not possible
-		"$@" --initialize-insecure --skip-ssl --datadir="$TMPDIR"
+		"$@" --initialize-insecure --datadir="$TMPDIR"
```
Comment (egegunes) [inline build/ps-entrypoint.sh:229]: does it work if user disables tls when creating a new cluster?

Comment (hors) [reply build/ps-entrypoint.sh:229]: but we do not support it 
https://github.com/percona/percona-server-mysql-operator/blob/main/deploy/cr.yaml#L9-L14
and if it does not work we will not have it for now

Comment (egegunes) [reply build/ps-entrypoint.sh:229]: true sorry confused with pxc

**Review by egegunes — APPROVED** [review]

**Pull request merged**